cloud/linode/instances_test.go

```diff
@@ -223,6 +215,65 @@ func TestMetadataRetrieval(t *testing.T) {
 				assert.Equal(t, meta.NodeAddresses, addresses)
 			}
 		})
+
+		getByIPTests := []struct {
+			name          string
+			nodeAddresses []v1.NodeAddress
+			expectedErr   error
+		}{
+			{name: "gets linode by External IP", nodeAddresses: []v1.NodeAddress{{
+				Type:    "ExternalIP",
+				Address: "172.234.31.123",
+			}, {
+				Type:    "InternalIP",
+				Address: "192.168.159.135",
+			}}},
+			{
+				name: "returns error on node with only internal IP", nodeAddresses: []v1.NodeAddress{{
+					Type:    "ExternalIP",
+					Address: "123.2.1.23",
+				}, {
+					Type:    "InternalIP",
+					Address: "192.168.159.135",
+				}},
+				expectedErr: cloudprovider.InstanceNotFound,
+			},
+			{
+				name: "returns error on no matching nodes by IP", nodeAddresses: []v1.NodeAddress{{
+					Type:    "ExternalIP",
+					Address: "123.2.1.23",
+				}, {
+					Type:    "InternalIP",
+					Address: "192.168.10.10",
+				}},
+				expectedErr: cloudprovider.InstanceNotFound,
+			},
+			{
+				name: "returns error on no node IPs", nodeAddresses: []v1.NodeAddress{},
+				expectedErr: fmt.Errorf("no IP address found on node test-node-1"),
+			},
+		}
+
+		for _, test := range getByIPTests {
+			t.Run(fmt.Sprintf("gets lindoe by IP - %s", test.name), func(t *testing.T) {
+				instances := newInstances(client)
+
+				publicIP := net.ParseIP("172.234.31.123")
```

Comment:
I'd probably hoist this out of the `t.Run` block, so that it's clear that this is static across all tests. This will only keep the dynamic parts of each test in the `t.Run` and make this slightly more readable.